### PR TITLE
Captions and Amazon support

### PR DESCRIPTION
**Pull request by johywang**
Changes made to the ExoMedia fork to support external captions. Implementation in Android-NewsApps will be done shortly.

TODO: Fine tune Amazon support to cover pre-Fire OS 5 and non-Fire TV.

### CODE REVIEW

File: library/src/main/java/com/devbrackets/android/exomedia/EMVideoView.java [hunk 1]
```diff
@@ -107,6 +108,7 @@
     private EMEventBus bus;
 
     private Uri videoUri;
+	private String captionsUri;
```
Comment (jblebrun) [inline library/src/main/java/com/devbrackets/android/exomedia/EMVideoView.java]: whitespace

File: library/src/main/java/com/devbrackets/android/exomedia/EMVideoView.java [hunk 2]
```diff
@@ -779,29 +781,62 @@ public void setVideoURI(Uri uri) {
...
     /**
      * Sets the Uri location for the video to play
      *
      * @param uri              The video's Uri
      * @param renderBuilder    RenderBuilder that should be used
      */
-    public void setVideoURI(Uri uri, RenderBuilder renderBuilder) {
+    public void setVideoURI(Uri uri, RenderBuilder renderBuilder, String captionsUri) {
         videoUri = uri;
+		this.captionsUri = captionsUri;
```
Comment (jblebrun) [inline library/src/main/java/com/devbrackets/android/exomedia/EMVideoView.java]: tab/spaces editor thing?

**Comment by johywang** [comment]
Ugh, yeah I just realized also that they used whitespace instead of tabs.

File: library/src/main/java/com/devbrackets/android/exomedia/builder/DashRenderBuilder.java [hunk 3]
```diff
@@ -237,10 +251,16 @@ private void buildRenderers(DrmSessionManager drmSessionManager, boolean filterH
 
             //Create the Sample Source to be used by the Closed Captions Renderer
             DataSource dataSourceCC = new DefaultUriDataSource(context, bandwidthMeter, userAgent);
-            ChunkSource chunkSourceCC = new DashChunkSource(manifestFetcher, DefaultDashTrackSelector.newAudioInstance(), dataSourceCC,
```
Comment (jblebrun) [inline library/src/main/java/com/devbrackets/android/exomedia/builder/DashRenderBuilder.java]: is this to handle captions embedded in the media file?

**Comment by jblebrun** [comment]
So is the issue basically that exoplayer has the timed text support, but the exomedia wrapper wasn't exposing it? But they had handling for captions embedded in media files for some renderers?

**Comment by jblebrun** [comment]
lgtm with the whitespace fixed

File: library/src/main/java/com/devbrackets/android/exomedia/builder/DashRenderBuilder.java (hunk 3 above)
Comment (johywang) [inline library/src/main/java/com/devbrackets/android/exomedia/builder/DashRenderBuilder.java]: Yeah, or whatever it was supposed to be doing. It seems that some of our videos should have embedded captions, but I wasn't able to get any of them to work.

**Comment by johywang** [comment]
Yup pretty much. I'm not sure if their code was handling embedded captions or sidecars, but I had to essentially expose the TextTrackRenderer to allow for a separate captions file.

**Comment by johywang** [comment]
Using 4 spaces instead of tabs.

File: library/src/main/java/com/devbrackets/android/exomedia/builder/DashRenderBuilder.java (hunk 3 above)
Comment (nbcklim) [inline library/src/main/java/com/devbrackets/android/exomedia/builder/DashRenderBuilder.java]: John: would you paraphrase Daniel's comment here, for documentation?

Comment (johywang) [inline library/src/main/java/com/devbrackets/android/exomedia/builder/DashRenderBuilder.java]: Daniel says that there should be CEA 608/708 embedded captions in the video assets coming from the Dreamy feeds, although it is not ideal to be using them as captions could change but the videos don't get republished.

**Pull request merged**